scripts/ci/in_container/kubernetes/app/deploy_app.sh

```diff
@@ -0,0 +1,185 @@
+
+SED_COMMAND="sed"
+
+if [[ "${KUBERNETES_MODE}" == "persistent_mode" ]]; then
+    ${SED_COMMAND} -e "s/{{INIT_GIT_SYNC}}//g" \
+        "${TEMPLATE_DIRNAME}/airflow.template.yaml" >"${BUILD_DIRNAME}/airflow.yaml"
+else
+    ${SED_COMMAND} -e "/{{INIT_GIT_SYNC}}/{r ${TEMPLATE_DIRNAME}/init_git_sync.template.yaml" -e 'd}' \
+        "${TEMPLATE_DIRNAME}/airflow.template.yaml" >"${BUILD_DIRNAME}/airflow.yaml"
+fi
+${SED_COMMAND} -i "s|{{AIRFLOW_CI_KUBERNETES_IMAGE}}|${AIRFLOW_CI_KUBERNETES_IMAGE}|g" "${BUILD_DIRNAME}/airflow.yaml"
+
+${SED_COMMAND} -i "s|{{CONFIGMAP_GIT_REPO}}|${CONFIGMAP_GIT_REPO}|g" "${BUILD_DIRNAME}/airflow.yaml"
+${SED_COMMAND} -i "s|{{CONFIGMAP_BRANCH}}|${CONFIGMAP_BRANCH}|g" "${BUILD_DIRNAME}/airflow.yaml"
+${SED_COMMAND} -i "s|{{INIT_DAGS_VOLUME_NAME}}|${INIT_DAGS_VOLUME_NAME}|g" "${BUILD_DIRNAME}/airflow.yaml"
+${SED_COMMAND} -i "s|{{POD_AIRFLOW_DAGS_VOLUME_NAME}}|${POD_AIRFLOW_DAGS_VOLUME_NAME}|g" \
+    "${BUILD_DIRNAME}/airflow.yaml"
+
+${SED_COMMAND} "s|{{CONFIGMAP_DAGS_FOLDER}}|${CONFIGMAP_DAGS_FOLDER}|g" \
+    "${TEMPLATE_DIRNAME}/configmaps.template.yaml" >"${BUILD_DIRNAME}/configmaps.yaml"
+${SED_COMMAND} -i "s|{{CONFIGMAP_GIT_REPO}}|${CONFIGMAP_GIT_REPO}|g" "${BUILD_DIRNAME}/configmaps.yaml"
+${SED_COMMAND} -i "s|{{CONFIGMAP_BRANCH}}|${CONFIGMAP_BRANCH}|g" "${BUILD_DIRNAME}/configmaps.yaml"
+${SED_COMMAND} -i "s|{{CONFIGMAP_GIT_DAGS_FOLDER_MOUNT_POINT}}|${CONFIGMAP_GIT_DAGS_FOLDER_MOUNT_POINT}|g" \
+    "${BUILD_DIRNAME}/configmaps.yaml"
+${SED_COMMAND} -i "s|{{CONFIGMAP_DAGS_VOLUME_CLAIM}}|${CONFIGMAP_DAGS_VOLUME_CLAIM}|g" \
+    "${BUILD_DIRNAME}/configmaps.yaml"
+
+cat "${BUILD_DIRNAME}/airflow.yaml"
+cat "${BUILD_DIRNAME}/configmaps.yaml"
+
+kubectl delete -f "${MY_DIR}/postgres.yaml" || true
+kubectl delete -f "${BUILD_DIRNAME}/airflow.yaml" || true
+kubectl delete -f "${MY_DIR}/secrets.yaml" || true
+
+set -e
+
+kubectl apply -f "${MY_DIR}/secrets.yaml"
+kubectl apply -f "${BUILD_DIRNAME}/configmaps.yaml"
+kubectl apply -f "${MY_DIR}/postgres.yaml"
+kubectl apply -f "${MY_DIR}/volumes.yaml"
+kubectl apply -f "${BUILD_DIRNAME}/airflow.yaml"
+
+dump_logs() {
+    POD=$(kubectl get pods -o go-template --template '{{range .items}}{{.metadata.name}}{{"\n"}}{{end}}' | grep airflow | head -1)
+    echo "------- pod description -------"
+    kubectl describe pod "${POD}"
+    echo "------- webserver init container logs - init -------"
+    kubectl logs "${POD}" -c init || true
+    if [[ "${KUBERNETES_MODE}" != "persistent_mode" ]]; then
+        echo "------- webserver init container logs - git-sync-clone -------"
+        kubectl logs "${POD}" -c git-sync-clone || true
+    fi
+    echo "------- webserver logs -------"
+    kubectl logs "${POD}" -c webserver || true
+    echo "------- scheduler logs -------"
+    kubectl logs "${POD}" -c scheduler || true
+    echo "--------------"
+}
+
+set +x
+# wait for up to 10 minutes for everything to be deployed
+PODS_ARE_READY="0"
+for i in {1..150}; do
```

Comment:
It looks weird. Can we use the while loop?
```bash
START="$(date +"%s")"
TIMEOUT=5

while [[ "$(date +"%s")" -lt "$((START + TIMEOUT))" ]]; do
    date;
    sleep 1;
done
```

Reply:
I would leave it as it is for now. It's quite more readable and does give a lot more (150*4 = ~600 s timeout + overhead).